feat: skip type-only imports and exports

tpluscode opened the pull request:
This fixes #10 

I tried this in my project and worked fine. Unfortunately I don't know what changes are necessary to get the jest test to pass. 

As of `7.9` (babel/babel#11171) this should be supported but somehow the parser keeps complaining about the syntax

karlprieb commented:
Great PR 👍 Thank you @tpluscode 
So, about the failing tests I will take a look tomorrow, but if you are free you can start investigating here https://github.com/facebook/jest/issues/9611

Also I updated all dependencies on the master branch... maybe you could do a rebase and add `babel-jest` to test it 😄

tpluscode commented:
You think this is related to jest? I thought that the `babel.transformSync` is only up to babel to parse and do its thing. Jest is not transforming the test statements, it it?

I looks like you master already has latest jest and `@babel/parser`. I only thought that maybe `preset-typescript` has to be added to the test case somehow but I could not get that to work either

tpluscode commented:
@karlprieb I think I'm done here.

I refactored the plugin code back to what it was initially and added a clearer check for type-only declarations. The test cases are also separated as they require the syntax plugin and I did not want to mix those
